agent/taskresource/cgroup/control/cgroupv2_controller_linux.go

```diff
@@ -0,0 +1,145 @@
+//go:build linux
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package control
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/cihub/seelog"
+	cgroupsv2 "github.com/containerd/cgroups/v2"
+)
+
+const (
+	defaultCgroupv2Path = "/sys/fs/cgroup"
+	parentCgroupSlice   = "/"
+	// This PID is only used when creating a slice for an existing systemd process.
+	// When creating a "general slice" that will be used as a parent slice for docker
+	// containers, then we use a dummy PID of -1.
+	// see https://github.com/containerd/cgroups/blob/1df78138f1e1e6ee593db155c6b369466f577651/v2/manager.go#L732-L735
+	generalSlicePID int = -1
+)
+
+// controlv2 is used to implement the cgroup Control interface
+type controlv2 struct{}
+
+// Create creates a new cgroup based off the spec post validation
+func (c *controlv2) Create(cgroupSpec *Spec) error {
+	// Validate incoming spec
+	err := validateCgroupSpec(cgroupSpec)
+	if err != nil {
+		return fmt.Errorf("cgroupv2 create: failed to validate spec: %w", err)
+	}
+
+	cgroupPath := cgroupSpec.Root
+	seelog.Infof("Creating cgroup cgroupv2root=%s parentSlice=%s cgroupPath=%s", defaultCgroupv2Path, parentCgroupSlice, cgroupPath)
+
+	m, err := cgroupsv2.NewSystemd(parentCgroupSlice, cgroupPath, generalSlicePID, cgroupsv2.ToResources(cgroupSpec.Specs))
+	if err != nil {
+		return fmt.Errorf("cgroupv2 create: unable to create v2 manager cgroupPath=%s err=%s", cgroupPath, err)
```

Comment:
nit: should we leave error formatting to the part that logs the error? (i.e. the consumer of `Create`)

Reply:
like a naked `return err` or do you mean just with less formatting than is here?

Reply:
I was thinking we can just wrap the error using go's way, like so: `return fmt.Errorf("unable to create v2 manager: %w", err)`

Then it's up to the consumer what to log and in what format, since it has all the info required. For example:

```go
err := Create(cgroupSpec)
if err != ni {
   logger.Info("Error creating cgroup", logger.Fields {
        "root": cgroupSpec.Root,
         "error": err,
}) 
} 
```

One difference would be, for example, that if the logger is configured in json format, then the format for the fields is also json.

Reply:
OK, this has been updated